app/src/main/java/org/wikipedia/notifications/NotificationActivity.kt

```diff
@@ -495,24 +510,45 @@ class NotificationActivity : BaseActivity() {
         }
     }
 
-    private inner class NotificationSearchBarHolder constructor(view: View) : RecyclerView.ViewHolder(view) {
+    private inner class NotificationSearchBarHolder constructor(view: View) :
+        RecyclerView.ViewHolder(view) {
+        val notificationFilterButton: AppCompatImageView = itemView.findViewById(R.id.notification_filter_button)
+        val notificationFilterCountView: TextView = itemView.findViewById(R.id.notification_filter_count)
+
         init {
             (itemView as WikiCardView).setCardBackgroundColor(ResourceUtil.getThemedColor(this@NotificationActivity, R.attr.color_group_22))
-            val notificationFilterButton = itemView.findViewById<View>(R.id.notification_filter_button)
+
+            updateFilterIconAndCount()
 
             itemView.setOnClickListener {
                 if (actionMode == null) {
+                    funnel.logSearchClick()
                     actionMode = startSupportActionMode(searchActionModeCallback)
                     postprocessAndDisplay()
                 }
             }
 
             notificationFilterButton.setOnClickListener {
+                funnel.logFilterClick()
                 startActivity(NotificationsFilterActivity.newIntent(it.context))
             }
 
             FeedbackUtil.setButtonLongPressToast(notificationFilterButton)
         }
+
+        private fun updateFilterIconAndCount() {
+            val fullWikiAndTypeListSize = NotificationsFilterActivity.allWikisList().size + NotificationsFilterActivity.allTypesIdList().size
+            val delimitedFiltersSizeString = Prefs.notificationsFilterLanguageCodes.orEmpty().split(",").filter { it.isNotEmpty() }.size
+            val enabledFilters = fullWikiAndTypeListSize - delimitedFiltersSizeString
+            if (enabledFilters == 0 || Prefs.notificationsFilterLanguageCodes == null) {
```

Comment:
Can we use `.isNullOrEmpty()` on the `Prefs.notificationsFilterLanguageCodes`?

Reply:
This doesnt apply when empty @cooltey